javascript/ql/lib/semmle/javascript/dataflow/internal/InterProceduralTypeInference.qll

```diff
@@ -190,16 +190,32 @@ private VarAccess getOnlyAccess(FunctionDeclStmt fn, LocalVariable v) {
   result = unique(VarAccess acc | acc = v.getAnAccess())
 }
 
+private VarAccess getOnlyAccessToFunctionExpr(FunctionExpr fn, LocalVariable v) {
+  exists( DeclStmt st |
+    fn = st.(DeclStmt).getADecl().getInit() and 
+    v = st.(DeclStmt).getADecl().getBindingPattern().getVariable() and
+    result = unique(VarAccess acc | acc = v.getAnAccess())
+  )
+}
```

Comment:
```suggestion
private VarAccess getOnlyAccessToFunctionExpr(FunctionExpr fn, LocalVariable v) {
  exists(VariableDeclarator decl |
    fn = decl.getInit() and
    v = decl.getBindingPattern().getVariable() and
    result = unique(VarAccess acc | acc = v.getAnAccess())
  )
}
```

Two small issues here: 
1: A `DeclStmt` might declare multiple variables (e.g. `var i = 1, j = 2`), and with your code it is possible to consider the function from one declaration and the variable from another. You need to ensure that the two `getADecl()` are the same declaration (or use `VariableDeclarator` directly).    
2: You need to run the auto-formatter (in VSCode: Ctrl + Shift + p -> `Format Document`).   

My suggestion does both, it's up to you whether you want to use it.